src/keys.rs

```diff
@@ -65,10 +67,12 @@ impl<PRFH: Hasher, MSGH: Hasher> Key<PRFH, MSGH> {
             return Ok(pk);
         }
 
-        let p_seed = self.p_seed;
-        let (public_key, _) =
-            self.params
-                .compute_ladders(&p_seed, None, &self.secret_key, false, false)?;
+        let (public_key, _) = self.params.compute_ladders(
+            &self.p_seed,
+            None,
+            &self.secret_key,
+            ComputeLaddersMode::ComputePublicKey,
+        )?;
```

Comment:
if it is possible compute the public key in `new` or `generate`, so we don't need to use `Option` for the public key anymore

Reply:
updated